### PR TITLE
feat: current course will be reloaded after playmode is exited

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Before entering playmode the current course will be serialized and on exit recreated which immitates the same behavior than unity has with changes in the scene (which will be deleted on playmode exit). This behavior also helps us to prevent corrupting SceneObjectReferences and PropertyReferences which are caching objects in scene.

### CODE REVIEW

File: Editor/DefaultEditingStrategy.cs [hunk 1]
```diff
@@ -106,8 +109,13 @@ public void HandleCurrentCourseChanged(string courseName)
             }
 
             PlayerPrefs.SetString(GlobalEditorHandler.LastEditedCourseNameKey, courseName);
-            course = CourseAssetManager.Load(courseName);
+            LoadCourse(CourseAssetManager.Load(courseName));
+
+        }
```
Comment (Gusinuhe) [inline Editor/DefaultEditingStrategy.cs:114]:
```suggestion
            LoadCourse(CourseAssetManager.Load(courseName));
        }
```

File: Editor/GlobalEditorHandler.cs [hunk 2]
```diff
@@ -140,5 +140,15 @@ internal static void ProjectIsGoingToSave()
...
+
+        internal static void EnterPlayMode()
+        {
+            strategy.HandleEnterPlayMode();
+        }
+
+        internal static void ExitPlayMode()
+        {
+            strategy.HandleExitingPlayMode();
+        }
```
Comment (Gusinuhe) [inline Editor/GlobalEditorHandler.cs:152]: Documentation might help (optional)

**Comment by aleksei-korolev** [comment]
Don't we already reload the course because we have to reload assemblies?

**Comment by SimonTheSourcerer** [comment]
I dont think assemblies are reloaded when playmode is exited.

**Comment by Luphoxa** [comment]
The issue that is solved with this PR also happens when the training is reloaded by the template. 
Can this also be addressed in this PR or is this a template issue?

**Comment by Luphoxa** [comment]
Is this a "yes, it's a template problem" to my question?

**Comment by Gusinuhe** [comment]
> Is this a "yes, it's a template problem" to my question?

I'll take a look.

**Comment by Gusinuhe** [comment]
Due to the nature of the issue reported by @Luphoxa, we considered it as a new bug.
A follow-up ticket was created:  [TRNG-1000](https://jira.innoactive.de/browse/TRNG-1000)

**Pull request merged**